src/status_im2/contexts/chat/messages/list/new_temp_view.cljs

```diff
@@ -150,22 +150,27 @@
         {:position :absolute
          :bottom   (+ (:bottom insets) composer.constants/composer-default-height 6)}]])))
 
+(defn use-keyboard-visibility
+  []
+  (let [show-listener (atom nil)
+        hide-listener (atom nil)
+        shown?        (atom nil)]
+    (rn/use-effect
+     (fn []
+       (reset! show-listener
+         (.addListener rn/keyboard "keyboardWillShow" #(reset! shown? true)))
+       (reset! hide-listener
+         (.addListener rn/keyboard "keyboardWillHide" #(reset! shown? false)))
+       (fn []
+         (.remove ^js @show-listener)
+         (.remove ^js @hide-listener))))
+    {:shown? shown?}))
+
+(defn- f-messages-list
+  [chat insets]
+  (let [{keyboard-shown? :shown?} (use-keyboard-visibility)]
```

Comment:
could you please elaborate why `use-keyboard` hook doean't work, and should we move this `use-keyboard-visibility` to react-native namespace so it could be used in other places ?

Reply:
When using keyboard hook, value of `keyboard-shown` does not update in `user-message-content` where we are reading the `keyboard-shown` value.

> should we move this use-keyboard-visibility to react-native namespace 
Probably no need. This is an edge case.

Reply:
i think it was because of the wrong usage of `:f>`, now `use-keyboard` hook should work properly

Reply:
I just tried using the `use-keyboard` hook again. Still the value of `keyboard-shown` does not update in `user-message-content`

Reply:
but why? is it registered bug in react native ?

Reply:
ok i see, its something with flat-list render data, ok, we should investigate this in the separate task

Reply:
Yes, it has to do with flat-list probably. The `keyboard-shown` value updates in the parent component, but does not get passed to the children. When using listeners and resetting the value on an atom it works.